packages/usa-date-picker/src/test/date-picker-i18n.spec.js

```diff
@@ -0,0 +1,182 @@
+const fs = require("fs");
+const path = require("path");
+const assert = require("assert");
+const EVENTS = require("./events");
+const DatePicker = require("../index");
+
+const TEMPLATE = fs.readFileSync(
+  path.join(__dirname, "/date-picker-i18n.template.html")
+);
+
+const datePickerSelector = () => document.querySelector(".usa-date-picker");
+
+const tests = [
+  { name: "document.body", selector: () => document.body },
+  { name: "date picker", selector: datePickerSelector },
+];
+
+tests.forEach(({ name, selector: containerSelector }) => {
+  describe(`date picker in multiple languages ${name}`, () => {
+    const { body } = document;
+
+    let root;
+    let button;
+
+    function changeLanguage(lang) {
+      document.documentElement.lang = lang;
+    }
+
+    beforeEach(() => {
+      body.innerHTML = TEMPLATE;
+      DatePicker.on(containerSelector());
+      root = datePickerSelector();
+      button = root.querySelector(".usa-date-picker__button");
+    });
+
+    afterEach(() => {
+      DatePicker.off(containerSelector());
+      window.onerror = null;
+      body.textContent = "";
+      changeLanguage("");
+    });
+
+    it("should display month in english by default", () => {
+      EVENTS.click(button);
+
+      const selectedMonth = root.querySelector(
+        ".usa-date-picker__calendar__month-selection"
+      );
+
+      assert.strictEqual(selectedMonth.innerHTML, "May");
+    });
+
+    it("should display month in the document language", () => {
+      changeLanguage("es");
+
+      EVENTS.click(button);
+
+      const selectedMonth = root.querySelector(
+        ".usa-date-picker__calendar__month-selection"
+      );
+
+      assert.strictEqual(selectedMonth.innerHTML, "mayo");
+    });
+
+    it("should display the correct aria label in the document language", () => {
+      changeLanguage("es");
+
+      EVENTS.click(button);
+
+      const selectedMonth = root.querySelector(
+        ".usa-date-picker__calendar__month-selection"
+      );
+
+      assert.strictEqual(
+        selectedMonth.getAttribute("aria-label"),
+        "mayo. Select month"
+      );
+    });
+
+    it("should display the full list of months in english by default", () => {
+      EVENTS.click(button);
+
+      const selectedMonth = root.querySelector(
+        ".usa-date-picker__calendar__month-selection"
+      );
+
+      EVENTS.click(selectedMonth);
+
+      const months = Array.from(
+        root.querySelectorAll(".usa-date-picker__calendar__table button")
+      ).map((btn) => btn.innerHTML);
```

Comment:
**Question:** This is a bit fine tuning but would it be possible to turn this into a function that we can call since we build this array in multiple tests? 

Pseudocode example:

```js
const weekdateSelecotr = ".usa-date-picker__calendar__day-of-week"
const monthSelector = ".usa-date-picker__calendar__table button"

cionst buildOptionsArray = (selector) => {
  Array.from(
    root.querySelectorAll(selector)
  ).map((btn) => btn.innerHTML);
};

// later

const months = buildOptionsArray(monthSelector);

// or

const daysOfWeek = buildOptionsArray(weekdaySelector);
```

Might need some additional work given the last test looks at the `aria-label` attribute rather than `innerHTML`.

Reply:
Personally I prefer keeping as much code for a particular test with that test. I don't think DRY should apply to test code.